implementation/src/main/java/io/smallrye/config/ConfigValueConfigSource.java

```diff
@@ -0,0 +1,39 @@
+package io.smallrye.config;
+
+import static java.util.stream.Collectors.toMap;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+public interface ConfigValueConfigSource extends ConfigSource {
+    ConfigValue getConfigValue(String propertyName);
+
+    Map<String, ConfigValue> getConfigValueProperties();
+
+    @Override
+    default Map<String, String> getProperties() {
+        return getConfigValueProperties().entrySet()
```

Comment:
This is likely to be very expensive... perhaps it would be better to return a map that is a view of the map returned by the delegate.